license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/header/HeaderDefinition.java

```diff
@@ -34,6 +36,7 @@
 
   private Pattern skipLinePattern;
   private Pattern firstLineDetectionPattern;
+  private Pattern otherLineDetectionPattern;
```

Comment:
I would prefer that the name is called `middleLineDetectionPattern`

Reply:
Except it is for all lines except the first line - the last line must also pass this pattern (or the beforeEachLine test).
Finding the best name is often the most difficult task ;-)

Reply:
> the last line must also pass this pattern

Why ? There is a pattern for the last line.

Reply:
But the code still checks beforeEachLine. I didn't want to change the logic and the RegEx happens when the startsWith(beforeEachLine) check happens.